### ISSUE TITLE
Question about benchmark

### ISSUE THREAD

**Issue by Kitsunetic** [yaorz97/PARENet#6]
Thanks for sharing great work! It seems this repository has much clearly arrange source code than most other papers in this domain!

I run reproduced test.py and eval.py with pretrained model on 3DMatch benchmark. However, the result looks strange. FMR and RR are too high and IR is too low.
Is this right? or, do I take mistake in somewhere?

```log
[2025-01-20 18:43:42] [CRIT]   Coarse Matching, PIR: 0.855, PMR>0: 0.997, PMR>=0.1: 0.986, PMR>=0.3: 0.961, PMR>=0.5: 0.915
[2025-01-20 18:43:42] [CRIT]     Kitchen, PIR: 0.897, PMR>0: 1.000, PMR>=0.1: 1.000, PMR>=0.3: 0.996, PMR>=0.5: 0.974
[2025-01-20 18:43:42] [CRIT]     Home_1, PIR: 0.887, PMR>0: 1.000, PMR>=0.1: 0.994, PMR>=0.3: 0.987, PMR>=0.5: 0.981
[2025-01-20 18:43:42] [CRIT]     Home_2, PIR: 0.720, PMR>0: 1.000, PMR>=0.1: 0.971, PMR>=0.3: 0.899, PMR>=0.5: 0.755
[2025-01-20 18:43:42] [CRIT]     Hotel_1, PIR: 0.927, PMR>0: 1.000, PMR>=0.1: 1.000, PMR>=0.3: 0.996, PMR>=0.5: 0.996
[2025-01-20 18:43:42] [CRIT]     Hotel_2, PIR: 0.877, PMR>0: 1.000, PMR>=0.1: 1.000, PMR>=0.3: 0.962, PMR>=0.5: 0.923
[2025-01-20 18:43:42] [CRIT]     Hotel_3, PIR: 0.883, PMR>0: 1.000, PMR>=0.1: 0.981, PMR>=0.3: 0.981, PMR>=0.5: 0.944
[2025-01-20 18:43:42] [CRIT]     Study, PIR: 0.835, PMR>0: 0.979, PMR>=0.1: 0.955, PMR>=0.3: 0.935, PMR>=0.5: 0.880
[2025-01-20 18:43:42] [CRIT]     MIT_Lab, PIR: 0.816, PMR>0: 1.000, PMR>=0.1: 0.987, PMR>=0.3: 0.935, PMR>=0.5: 0.870
[2025-01-20 18:43:42] [CRIT]   Fine Matching, FMR: 1.000, IR: 0.699, OV: 0.970, std: 0.000
[2025-01-20 18:43:42] [CRIT]     Kitchen, FMR: 1.000, IR: 0.705
[2025-01-20 18:43:42] [CRIT]     Home_1, FMR: 1.000, IR: 0.713
[2025-01-20 18:43:42] [CRIT]     Home_2, FMR: 1.000, IR: 0.701
[2025-01-20 18:43:42] [CRIT]     Hotel_1, FMR: 1.000, IR: 0.701
[2025-01-20 18:43:42] [CRIT]     Hotel_2, FMR: 1.000, IR: 0.711
[2025-01-20 18:43:42] [CRIT]     Hotel_3, FMR: 1.000, IR: 0.710
[2025-01-20 18:43:42] [CRIT]     Study, FMR: 1.000, IR: 0.692
[2025-01-20 18:43:42] [CRIT]     MIT_Lab, FMR: 1.000, IR: 0.663
[2025-01-20 18:43:42] [CRIT]   Registration, RR: 0.991, mean_RRE: 1.520, mean_RTE: 0.050, median_RRE: 1.357, median_RTE: 0.041
[2025-01-20 18:43:42] [CRIT]     Kitchen, RR: 1.000, mean_RRE: 2.128, mean_RTE: 0.034, median_RRE: 2.116, median_RTE: 0.030
[2025-01-20 18:43:42] [CRIT]     Home_1, RR: 1.000, mean_RRE: 1.176, mean_RTE: 0.039, median_RRE: 1.049, median_RTE: 0.037
[2025-01-20 18:43:42] [CRIT]     Home_2, RR: 0.925, mean_RRE: 1.990, mean_RTE: 0.073, median_RRE: 1.681, median_RTE: 0.049
[2025-01-20 18:43:42] [CRIT]     Hotel_1, RR: 1.000, mean_RRE: 1.344, mean_RTE: 0.047, median_RRE: 1.241, median_RTE: 0.043
[2025-01-20 18:43:42] [CRIT]     Hotel_2, RR: 1.000, mean_RRE: 1.129, mean_RTE: 0.043, median_RRE: 0.998, median_RTE: 0.042
[2025-01-20 18:43:42] [CRIT]     Hotel_3, RR: 1.000, mean_RRE: 1.212, mean_RTE: 0.041, median_RRE: 1.127, median_RTE: 0.036
[2025-01-20 18:43:42] [CRIT]     Study, RR: 1.000, mean_RRE: 1.759, mean_RTE: 0.067, median_RRE: 1.461, median_RTE: 0.057
[2025-01-20 18:43:42] [CRIT]     MIT_Lab, RR: 1.000, mean_RRE: 1.422, mean_RTE: 0.057, median_RRE: 1.182, median_RTE: 0.038
```

**Comment by yaorz97** [comment]
Hi, the result is not correct. Is there a problem with your prepared dataset? The correct result should be: 

![Image](https://github.com/user-attachments/assets/ead4f26d-a422-41d2-a4f7-df07897161cf)

**Comment by Kitsunetic** [comment]
Thanks for fast response!
I found the reason, it was my mistake while reproducing test.py.
I forgot making the model evaluation mode (`model.eval()`).
After modifying this, the result look like this:
```log
[2025-01-21 00:09:37] [CRIT]   Coarse Matching, PIR: 0.855, PMR>0: 0.997, PMR>=0.1: 0.984, PMR>=0.3: 0.962, PMR>=0.5: 0.917
[2025-01-21 00:09:37] [CRIT]     Kitchen, PIR: 0.898, PMR>0: 1.000, PMR>=0.1: 1.000, PMR>=0.3: 0.996, PMR>=0.5: 0.976
[2025-01-21 00:09:37] [CRIT]     Home_1, PIR: 0.888, PMR>0: 1.000, PMR>=0.1: 0.994, PMR>=0.3: 0.987, PMR>=0.5: 0.981
[2025-01-21 00:09:37] [CRIT]     Home_2, PIR: 0.719, PMR>0: 1.000, PMR>=0.1: 0.971, PMR>=0.3: 0.899, PMR>=0.5: 0.755
[2025-01-21 00:09:37] [CRIT]     Hotel_1, PIR: 0.926, PMR>0: 1.000, PMR>=0.1: 0.996, PMR>=0.3: 0.996, PMR>=0.5: 0.996
[2025-01-21 00:09:37] [CRIT]     Hotel_2, PIR: 0.877, PMR>0: 1.000, PMR>=0.1: 1.000, PMR>=0.3: 0.962, PMR>=0.5: 0.923
[2025-01-21 00:09:37] [CRIT]     Hotel_3, PIR: 0.882, PMR>0: 1.000, PMR>=0.1: 0.981, PMR>=0.3: 0.981, PMR>=0.5: 0.944
[2025-01-21 00:09:37] [CRIT]     Study, PIR: 0.839, PMR>0: 0.979, PMR>=0.1: 0.955, PMR>=0.3: 0.938, PMR>=0.5: 0.894
[2025-01-21 00:09:37] [CRIT]     MIT_Lab, PIR: 0.814, PMR>0: 1.000, PMR>=0.1: 0.974, PMR>=0.3: 0.935, PMR>=0.5: 0.870
[2025-01-21 00:09:37] [CRIT]   Fine Matching, FMR: 0.984, IR: 0.736, OV: 0.947, std: 0.013
[2025-01-21 00:09:37] [CRIT]     Kitchen, FMR: 0.998, IR: 0.763
[2025-01-21 00:09:37] [CRIT]     Home_1, FMR: 0.994, IR: 0.762
[2025-01-21 00:09:37] [CRIT]     Home_2, FMR: 0.971, IR: 0.630
[2025-01-21 00:09:37] [CRIT]     Hotel_1, FMR: 0.996, IR: 0.789
[2025-01-21 00:09:37] [CRIT]     Hotel_2, FMR: 0.990, IR: 0.753
[2025-01-21 00:09:37] [CRIT]     Hotel_3, FMR: 0.981, IR: 0.755
[2025-01-21 00:09:37] [CRIT]     Study, FMR: 0.955, IR: 0.729
[2025-01-21 00:09:37] [CRIT]     MIT_Lab, FMR: 0.987, IR: 0.710
[2025-01-21 00:09:37] [CRIT]   Registration, RR: 0.952, mean_RRE: 1.888, mean_RTE: 0.062, median_RRE: 1.616, median_RTE: 0.051
[2025-01-21 00:09:37] [CRIT]     Kitchen, RR: 0.996, mean_RRE: 2.366, mean_RTE: 0.043, median_RRE: 2.293, median_RTE: 0.038
[2025-01-21 00:09:37] [CRIT]     Home_1, RR: 0.972, mean_RRE: 1.500, mean_RTE: 0.049, median_RRE: 1.373, median_RTE: 0.042
[2025-01-21 00:09:37] [CRIT]     Home_2, RR: 0.836, mean_RRE: 2.282, mean_RTE: 0.082, median_RRE: 1.929, median_RTE: 0.062
[2025-01-21 00:09:37] [CRIT]     Hotel_1, RR: 0.995, mean_RRE: 1.622, mean_RTE: 0.058, median_RRE: 1.438, median_RTE: 0.051
[2025-01-21 00:09:37] [CRIT]     Hotel_2, RR: 0.962, mean_RRE: 1.575, mean_RTE: 0.060, median_RRE: 1.348, median_RTE: 0.049
[2025-01-21 00:09:37] [CRIT]     Hotel_3, RR: 0.962, mean_RRE: 1.547, mean_RTE: 0.048, median_RRE: 1.224, median_RTE: 0.047
[2025-01-21 00:09:37] [CRIT]     Study, RR: 0.915, mean_RRE: 2.190, mean_RTE: 0.079, median_RRE: 1.819, median_RTE: 0.062
[2025-01-21 00:09:37] [CRIT]     MIT_Lab, RR: 0.978, mean_RRE: 2.021, mean_RTE: 0.073, median_RRE: 1.500, median_RTE: 0.056
```

It looks fine, right?
